### PR TITLE
974 add right icon variant to ic button

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes

Add right icon slot, deprecate icon slot and create left icon slot, and update icon sizes and gap.

## Related issue
#974 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Comment by ad9242** [comment]
the icons on top navigation popout menu appear huge!
image

File: packages/web-components/src/components/ic-navigation-button/ic-navigation-button.tsx [hunk 1]
```diff
@@ -162,7 +162,11 @@ export class NavigationButton {
           {...this.inheritedAttributes}
         >
           {label}
-          <slot slot="icon" name="icon"></slot>
+          {variant === "icon" ? (
+            <slot name="icon"></slot>
```
Comment (ad9242) [inline packages/web-components/src/components/ic-navigation-button/ic-navigation-button.tsx:166]: i think the `slot="icon"` is needed to tell it which slot in ic-button to use. Should it be `slot="left-icon"` now?

Comment (GCHQ-Developer-530) [reply packages/web-components/src/components/ic-navigation-button/ic-navigation-button.tsx:166]: Yeah I think that fixes it. Will update now 👍

File: packages/web-components/src/components/ic-button/ic-button.tsx [hunk 2]
```diff
@@ -33,7 +33,9 @@ import {
 let buttonIds = 0;
 
 /**
- * @slot icon - Content will be placed to the left of the button label.
+ * @slot icon - Deprecated. Use left-icon or right-icon slot instead.
```
[the remark this answers is not in the document]
Comment (GCHQ-Developer-530) [reply packages/web-components/src/components/ic-button/ic-button.tsx:36]: Yeah it's a difficult one, I tried adding the @deprecated in a couple of places and it seemed to think the entire component was deprecated in the readme, which wasn't ideal. I'll add the slot should not be used anymore sentence to it, but not sure there's a solution other than adding deprecated as a regular word.

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Pull request merged**